src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -314,7 +346,19 @@ export class MeetingManager implements AudioVideoObserver {
 
   async listAndSelectDevices(): Promise<void> {
     await this.updateDeviceLists();
+
+    let hasAudioInput = false, hasAudioOutput = false, hasVideoInput = false;
+    const devices = await navigator.mediaDevices.enumerateDevices();
+    for (const device of devices) {
+      if (device.label) {
+        if (device.kind === "audioinput") hasAudioInput = true;
+        if (device.kind === "audiooutput") hasAudioOutput = true;
+        if (device.kind === "videoinput") hasVideoInput = true;
+      }
+    }
+
     if (
+      hasAudioInput &&
```

Comment:
@richnew10 This is to handle the error and enhance user experience. 
In `listAndSelectDevices` function, we call `audioVideo?.choose*Device()` which may call `getUserMedia()`. This could lead to some errors which users don't expect, after we merged this PR.

For example, user wants to join in view-only mode. They pass the `deviceLabels.None` via `mm.join(deviceLabels)`. This will set the `deviceLabelTrigger` as `() => Promise.resolve(new MediaStream());`. If we don't make this change, when user joins the meeting, the devices with empty labels will be selected and the below errors pops up:
![image](https://user-images.githubusercontent.com/20471438/121099636-10335000-c7ad-11eb-9d86-430590cb90de.png)

This seems a limit of our SDK. Making this change could address this issue.

Reply:
@richnew10 
There are two reason why I made it like this:
1. Enhance user experience
If user joins a meeting without a kind of device, user may expect the 'None' is selected for this kind of device.
Without this change, the device with empty label (it is the first one) may be selected. I thought this could be a bad behavior.

2. Avoid error
It seems the error is **only about audio device**. Whenever user ignore the audio device(`DeviceLabels.Video`, `DeviceLabels.None`), without this change, the above error pops up in Chrome, Firefox and Safari. 
I think if we support user to select the device kind they want to trigger device permission prompt, we shouldn't show any related error.
Besides, the Firefox will still trigger the device permission prompt for audio device only. This seems to be caused by `chooseAudioInputDevice` => `chooseInputIntrinsicDevice()` => 
    ```ts
    const proposedConstraints = this.calculateMediaStreamConstraints(kind, device);
    newDevice.stream = await navigator.mediaDevices.getUserMedia(proposedConstraints);
    ```
    You can check the code [here](https://github.com/aws/amazon-chime-sdk-js/blob/952ad05d94980aeff09693ab2e9efd6df6066661/src/devicecontroller/DefaultDeviceController.ts#L1396). I don't know how to address this issue.

Reply:
There is clearly a bug here, and it might be related to the misordered init we've observed. Let's follow up on this.

Reply:
Recorded the bug. I will keep my eye on it.